astropy/io/ascii/fixedwidth.py

```diff
@@ -262,10 +262,7 @@ def write(self, lines):
         header_rows = getattr(self, "header_rows", default_header_rows)
         # First part is getting the widths of each column.
         # List (rows) of list (column values) for data lines
-        vals_list = []
-        col_str_iters = self.str_vals()
-        for vals in zip(*col_str_iters):
-            vals_list.append(vals)
+        vals_list = list(zip(*self.str_vals()))
```

Comment:
Why was this change (and two others like it) made? The original is written that way to make the code somewhat easier to understand.

Reply:
Partly speed. This way is 2x faster on newer Python. Partly readability. IMO this one-liner is more clear. What was 4 lines and included intermediate variables is now 3 sequential function calls on one line. It's more clearly a single thought and it's short enough to parse in a glance.

Reply:
I also agree that the new code is more Pythonic and easier to read but the initial read does take time to parse if you are not familiar with the chaining.

Reply:
I agree this is a good change, we should keep the updated code.

Reply:
Is readability a directive in Ruff anyway? ;-)
But in this case I agree as well the one-liner is rather easier to parse; it's mainly identifying the role of `self.str_vals()` I got stuck on, but that cannot be helped by either format.